tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -260,7 +286,7 @@ public function testColumnCharsetChange(): void
         $diff = $fromSchema->getMigrateToSql($toSchema, $this->connection->getDatabasePlatform());
         self::assertContains(
             'ALTER TABLE test_column_charset_change CHANGE col_string'
-                . ' col_string VARCHAR(100) CHARACTER SET ascii NOT NULL',
+                . ' col_string VARCHAR(100) CHARACTER SET ascii NOT NULL FIRST',
```

Comment:
Why do we expect the SQL to contain the `FIRST`/`AFTER` qualifiers if the column position doesn't change? I believe these should be only generated when necessary.

Reply:
we generate this for ALTER TABLE only, for TABLE CREATE, position is not generated

if position is the same, we regenerate whole column definition

Reply:
Can we generate the column position only if it's changed?

Reply:
I can add a condition, if the previous column is the same for fromTable and toTable, then to not add the position. Tests may however need more changes as currently, we expect that SQL column definition is always fully regenerated.

Let solve this as the last step before merge, firstly, I want to make sure `toTable` property will stay

Reply:
I think `toTable` should stay since the other solution with `ColumnDiff` instance will not handle the case of new columns, which means we can resume this conversation. Let's generate the position only when necessary please.

Reply:
> Let's generate the position only when necessary please.

I understand we want to generate minimalistic diff, but I think we do not want it. ALTER TABLE ADD/MODIFY column is always generated fully even if there is a minor column definition change like column comment. Adding position based on the original table will violate this definition.

Reply:
@morozov @greg0ire can we agree on this and can we merge this PR?

Reply:
> can we agree on this

If it's always generated fully indeed then it's ok with me.

> can we merge this PR

In theory yes, in practice the tests fail

Reply:
CI fixed